gcp-ts-docker-gcr-cloudrun/cloud-run-deploy/index.ts

```diff
@@ -0,0 +1,59 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as gcp from "@pulumi/gcp";
+import * as docker from "@pulumi/docker";
+
+const location = gcp.config.region || "us-central1";
+
+const config = new pulumi.Config()
+
+const configFile = config.require("docker-config-file")
+
+// Default to using ruby-app since that is the name of the image for the example.    
+const imageName = config.get("image-name") ?? "ruby-app"
+
+const gcrDockerProvider = new docker.Provider('gcr', {
+    registryAuth: [{
+        address: "gcr.io",
+        configFile: configFile
+    }],
+})
+
+// Used to get the image from the google cloud registry.  Output is required to make sure that the provider is in sync with this call. 
+const registryImage = pulumi.output(
+    docker.getRegistryImage({
+    name: `gcr.io/${gcp.config.project}/${imageName}:latest`,
+}, {provider: gcrDockerProvider}));
+
+
+// Using the value from the registryImage to pull the image if it's new, pullTriggers looks for a new sha. 
+var dockerImage = registryImage.apply(r => new docker.RemoteImage(`${imageName}-docker-image`, {
```

Comment:
Do you need an `apply` here? Could you do `registryImage.name` the same way you do `registryImage.sha256Digest`?

Reply:
Yeah that's why I added the apply, weirdly got an error when trying that...